File: cpp ref/Meeple.cpp
#include "Meeple.h"

Meeple::Meeple(int ownerId){
	this->ownerId = ownerId;
	this->status = 0;
	location.x = -1;
	location.y = -1;
	return;
}

void Meeple::updateStatus(int status){
	this->status = status;
	return;
}

void Meeple::updateLocation(int x, int y){
	location.x = x;
	location.y = y;
}

int Meeple::getOwnerId(){
	return ownerId;
}

int Meeple::getLocationX(){
	return location.x;
}

int Meeple::getLocationY(){
	return location.y;
}

int Meeple::getStatus(){
	return status;
}

Meeple::~Meeple() {

	return;
}

File: Meeple.java

public class Meeple {
	int owner;
	int ID;
	MeepleStatusEnum status;
	Location location;
	
	public Meeple(){
		this.owner = -1;
		this.status = MeepleStatusEnum.onNone;
		this.location = new Location(-1,-1);
		this.ID = 0;
	}
	
	public Meeple(int owner, int ID){
		this.owner = owner;
		this.status = MeepleStatusEnum.onNone;
		this.location = new Location(-1, -1);
		this.ID = ID;
	}
	
	void updateLocation(Location location){
		this.location = location;
	}
	
	void setStatus(int newStatus){
		this.status = convertIntToMeepleStatus(newStatus);
	}

	void setStatus(MeepleStatusEnum newStatus){
		this.status = newStatus;
	}
	
	int getOwner(){
		return owner;
	}
	
	Location getLocation() { return location; }

	MeepleStatusEnum getStatus(){
		return status;
	}

	public MeepleStatusEnum convertIntToMeepleStatus (int status){
		switch (status) {
			case 0:
				return MeepleStatusEnum.onField;
			case 1:
			case 4:
				return MeepleStatusEnum.onRoad;
			case 2:
			case 3:
			case 5:
				return MeepleStatusEnum.onCity;
			case 6:
				return MeepleStatusEnum.onMonastery;
			case 9:
				return MeepleStatusEnum.onNone;
			default:
				throw new IllegalStateException();

		}
	}
}